test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -698,6 +698,49 @@ TEST_F(StatisticsDomainParticipantTests, EnableStatisticsDataWriterFailureIncomp
 #endif // FASTDDS_STATISTICS
 }
 
+/**
+
+ * This test checks that a participant is correctly deleted by the factory
+ * after calling delete_contained_entities() method
+ * 1. Create a statistics participant, register type
+ * 2. Create a sample topic
+ * 3. Perform a delete_contained_entities() in the statistics participant
+ * 4. Delete the participant
+ */
+TEST_F(StatisticsDomainParticipantTests, DeleteParticipantAfterDeleteContainedEntitiesFailure)
+{
+#ifdef FASTDDS_STATISTICS
+
+    eprosima::fastdds::dds::TypeSupport count_type(new EntityCountPubSubType);
+
+    // 1. Create DomainParticipant with statistics
+    eprosima::fastdds::dds::DomainParticipant* participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+                    create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+    // Register type
+    participant->register_type(count_type);
+
+    // 2. Create a sample topic
+    participant->create_topic(HEARTBEAT_COUNT_TOPIC,
+            count_type->getName(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+
+    DomainParticipant* statistics_participant = DomainParticipant::narrow(participant);
+    ASSERT_NE(statistics_participant, nullptr);
+
+    EXPECT_EQ(ReturnCode_t::RETCODE_OK, statistics_participant->enable_statistics_datawriter(HEARTBEAT_COUNT_TOPIC,
+            STATISTICS_DATAWRITER_QOS));
+
+    // 3. Perform a delete_contained_entities() in the statistics participant
+    EXPECT_EQ(statistics_participant->delete_contained_entities(), ReturnCode_t::RETCODE_OK);
```

Comment:
I would call this method from `participant` and not from `statistics_participant`, as this would be the most expected way to do it from a user. (It should be the same, but just in case, to avoid virtualization method problems).